src/cas/flatten_data_to_tables.py

```diff
@@ -0,0 +1,257 @@
+                record["cell_fullname"] = ""
+                record["parent_cell_set_accession"] = ""
+                record["parent_cell_set_name"] = ""
+                record["labelset"] = str(annotation_object.get("labelset", "")).replace("_name", "")
+                std_parent_records.append(record)
+        if "parent_cell_set_name" in annotation_object:
+            record["parent_cell_set_name"] = annotation_object["parent_cell_set_name"]
+            if annotation_object["parent_cell_set_name"] in std_parent_records_dict:
+                std_parent_records_dict.get(annotation_object["parent_cell_set_name"]).append(record)
+            else:
+                children = list()
+                children.append(record)
+                std_parent_records_dict[annotation_object["parent_cell_set_name"]] = children
+        if "parent_cell_set_accession" in annotation_object:
+            record["parent_cell_set_accession"] = annotation_object["parent_cell_set_accession"]
+    assign_parent_accession_ids(accession_manager, std_parent_records, std_parent_records_dict, cta["labelsets"])
+    std_records.extend(std_parent_records)
+    std_records_df = pd.DataFrame.from_records(std_records)
+    std_records_df.to_csv(std_data_path, sep="\t", index=False)
+    return std_data_path
+
+
+def list_to_string(my_list: list):
+    """
+    Converts a list to its string representation. Nanobot has problem with single quotations so removes them as well.
+    Parameters:
+        my_list: list to serialize
+    Returns:
+        string representation of the list
+    """
+    if not my_list:
+        str_value = "[]"
+    else:
+        str_value = str(my_list).replace("'", "\"")
+    return str_value
+
+
+def assign_parent_accession_ids(accession_manager, std_parent_records, std_parent_records_dict, labelsets):
+    """
+    Assigns accession ids to parent clusters and updates their references from the child clusters.
+    Parameters:
+        accession_manager: accession ID generator
+        std_parent_records: list of all parents to assign accession ids
+        std_parent_records_dict: parent cluster - child clusters dictionary
+        labelsets: labelsets list
+    """
+    label_set_ranks = dict([(label_set["name"].replace("_name", ""), label_set["rank"]) for label_set in labelsets])
+
+    std_parent_records.sort(key=lambda x: int(label_set_ranks[x["labelset"]]))
+    for std_parent_record in std_parent_records:
+        accession_id = accession_manager.generate_accession_id()
+        std_parent_record["cell_set_accession"] = accession_id
+
+        children = std_parent_records_dict.get(std_parent_record["cell_label"], list())
+        for child in children:
+            child["parent_cell_set_accession"] = accession_id
+
+
+def normalize_column_name(column_name: str) -> str:
+    """
+    Normalizes column name for nanobot compatibility.
+    Nanobot column name requirement: All names must match: ^[\w_ ]+$' for to_url()
+
```

Comment:
Maybe we could push this upstream to validation of input spreadhseet?  (Just noting here.  Should be in separate ticket/PR)